tools/install_builder_prerequisites.sh

```diff
@@ -71,57 +102,62 @@ function confirm
-  )
-  sudo apt-get update
-  if [[ "${ASSUME_YES}" == "true" ]]; then
-    sudo apt-get install -y ${packages[@]}
+
+function install_packages {
+  if [[ "${HOST_OS}" == "Linux" ]]; then
+    packages=(
+      'libpq-dev'
+      'pkg-config'
+      'xmlsec1'
+      'libxmlsec1-dev'
+      'libxmlsec1-openssl'
+      'libclang-dev'
+      'libsqlite3-dev'
+    )
+    sudo apt-get update
+    if [[ "${ASSUME_YES}" == "true" ]]; then
+        sudo apt-get install -y "${packages[@]}"
+    else
+        confirm "Install (or update) [${packages[*]}]?" && sudo apt-get install "${packages[@]}"
+    fi
+  elif [[ "${HOST_OS}" == "SunOS" ]]; then
+    packages=(
+      'pkg:/package/pkg'
+      'build-essential'
+      'library/postgresql-13'
+      'pkg-config'
+      'library/libxmlsec1'
+      # "bindgen leverages libclang to preprocess, parse, and type check C and C++ header files."
+      'pkg:/ooce/developer/clang-120'
+    )
+
+    # Install/update the set of packages.
+    # Explicitly manage the return code using "rc" to observe the result of this
+    # command without exiting the script entirely (due to bash's "errexit").
+    rc=0
+    confirm "Install (or update) [${packages[*]}]?" && { pfexec pkg install -v "${packages[@]}" || rc=$?; }
+    # Return codes:
+    #  0: Normal Success
+    #  4: Failure because we're already up-to-date. Also acceptable.
+    if [[ "$rc" -ne 4 ]] && [[ "$rc" -ne 0 ]]; then
+      exit "$rc"
+    fi
+
+    pkg list -v "${packages[@]}"
+  elif [[ "${HOST_OS}" == "Darwin" ]]; then
+    packages=(
+      'postgresql'
+      'pkg-config'
+      'libxmlsec1'
+    )
+    confirm "Install (or update) [${packages[*]}]?" && brew install "${packages[@]}"
   else
-      confirm "Install (or update) [${packages[*]}]?" && sudo apt-get install ${packages[@]}
-  fi
-elif [[ "${HOST_OS}" == "SunOS" ]]; then
-  packages=(
-    'pkg:/package/pkg'
-    'build-essential'
-    'library/postgresql-13'
-    'pkg-config'
-    'library/libxmlsec1'
-    # "bindgen leverages libclang to preprocess, parse, and type check C and C++ header files."
-    'pkg:/ooce/developer/clang-120'
-  )
-
-  # Install/update the set of packages.
-  # Explicitly manage the return code using "rc" to observe the result of this
-  # command without exiting the script entirely (due to bash's "errexit").
-  rc=0
-  confirm "Install (or update) [${packages[*]}]?" && { pfexec pkg install -v "${packages[@]}" || rc=$?; }
-  # Return codes:
-  #  0: Normal Success
-  #  4: Failure because we're already up-to-date. Also acceptable.
-  if [[ "$rc" -ne 4 ]] && [[ "$rc" -ne 0 ]]; then
-    exit "$rc"
+    echo "Unsupported OS: ${HOST_OS}"
+    exit 1
   fi
+}
 
-  pkg list -v "${packages[@]}"
-elif [[ "${HOST_OS}" == "Darwin" ]]; then
-  packages=(
-    'postgresql'
-    'pkg-config'
-    'libxmlsec1'
-  )
-  confirm "Install (or update) [${packages[*]}]?" && brew install ${packages[@]}
-else
-  echo "Unsupported OS: ${HOST_OS}"
-  exit -1
-fi
+install_packages
```

Comment:
Do we want to `retry` this too? I know we've seen failures in CI when `apt`-installing things (although maybe that's fixed now with the change of apt source?).

Reply:
Sure, I'll do the same for the `install_runner_prerequisites.sh` script too.